common/include/pcl/common/impl/common.hpp

```diff
@@ -70,6 +70,20 @@ pcl::getAngle3D (const Eigen::Vector3f &v1, const Eigen::Vector3f &v2, const boo
 inline void
 pcl::getMeanStd (const std::vector<float> &values, double &mean, double &stddev)
 {
+    try
+    {
+	    // throw a exception when there are less than two elements in vector `values` 
+	    // to prevent division by zero error while calculating standard deviation.
+	    if(values.size() < 2 )
+	    {
+		throw std::length_error("Input array must have atleast 2 elements.");
+	    }
+    } 
+    catch(length_error& e)
```

Comment:
What's the point of throwing an exception and immediately catching it?

Reply:
I did not know if any top-level hierarchy is catching any exceptions or not, so I had to catch it here itself. Although I did forget to add a return statement after printing the error message. I'll add it.
But even that leaves the function call in a weird result state. What will the user expect when he passes an empty array? Should I find where this function is called and add a catch block there itself?
Something else on your mind?

Reply:
The idea it to let it blow up for the user. That way he'll need to sanitize `values` before passing it to this method.

Reply:
oh. If you want to blow up, why throw an exception at all? Aren't exceptions caused to be handled?
Anyway, I'll assume you have a use case in mind for expecting such a behavior.
So, in short,
You want to display a message saying `atleast 2 values` and then crash, yes?

Reply:
> oh. If you want to blow up, why throw an exception at all? Aren't exceptions caused to be handled?

Yes but, to be handled by the user ('s code). We have no way of guessing what he wants to do in case he doesn't have a sanitized input. We just want to make sure that he is explicitly dealing with it. 

> You want to display a message saying atleast 2 values and then crash, yes?

Yes.